packages/cloudflare/src/cli/templates/worker.ts

```diff
@@ -9,23 +9,19 @@ import { MockedResponse } from "next/dist/server/lib/mock-request";
 import type { NodeRequestHandler } from "next/dist/server/next-server";
 
 import type { CloudflareContext } from "../../api";
+import { createALSProxy } from "./utils";
 
 const NON_BODY_RESPONSES = new Set([101, 204, 205, 304]);
 
+const processEnvALS = new AsyncLocalStorage<Record<string, unknown>>();
 const cloudflareContextALS = new AsyncLocalStorage<CloudflareContext>();
 
 // Note: this symbol needs to be kept in sync with the one defined in `src/api/get-cloudflare-context.ts`
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-(globalThis as any)[Symbol.for("__cloudflare-context__")] = new Proxy(
-  {},
-  {
-    ownKeys: () => Reflect.ownKeys(cloudflareContextALS.getStore()!),
-    getOwnPropertyDescriptor: (_, ...args) =>
-      Reflect.getOwnPropertyDescriptor(cloudflareContextALS.getStore()!, ...args),
-    get: (_, property) => Reflect.get(cloudflareContextALS.getStore()!, property),
-    set: (_, property, value) => Reflect.set(cloudflareContextALS.getStore()!, property, value),
-  }
-);
+(globalThis as any)[Symbol.for("__cloudflare-context__")] = createALSProxy(cloudflareContextALS);
+
+// @ts-expect-error - populated when we run inside the ALS context
+globalThis.process.env = createALSProxy(processEnvALS);
```

Comment:
I think I agree with @vicb to some extent here.
Before any request, there might already be values on `process.env`, which are being lost by assigning it here.
How about we capture the original `process.env` somehow and then include that in the initialization of the store below.

E.g.

```
const originalEnv = globalThis.process.env;
globalThis.process.env = createALSProxy(processEnvALS);
...
export default {
  async fetch(request, env, ctx) {
    return processEnvALS.run({ NODE_ENV: "production", ...originalEnv, ...env }, () => {
```